### PR TITLE
BenchmarkRunner option to generate query plan diagrams in DOT format

### PR DESCRIPTION
**Pull request by andygrove**
Signed-off-by: Andy Grove <andygrove@nvidia.com>

This PR adds a `--generate-dot` flag to the benchmark runner so that it will generate query plan diagrams in DOT format. The generator is also updated to color-code operators as green or blue depending on whether they are on GPU or CPU, with transitions highlighted in red. 

The chart is also now inverted with the final operator at the bottom, to match Spark UI.

colorful2

### CODE REVIEW

File: integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/BenchmarkRunner.scala [hunk 1]
```diff
@@ -89,23 +89,26 @@ object BenchmarkRunner {
                   path,
                   iterations = conf.iterations(),
                   summaryFilePrefix = summaryFilePrefixWithQuery,
-                  gcBetweenRuns = conf.gcBetweenRuns())
+                  gcBetweenRuns = conf.gcBetweenRuns(),
+                  generateDotGraph = conf.generateDot())
```
Comment (jlowe) [inline integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/BenchmarkRunner.scala:93]: Not a change that needs to be in this PR, but the number of options is getting to the point where we should consider using a builder pattern, building an options object and passing that.

File: integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/BenchUtils.scala [hunk 2]
```diff
@@ -448,8 +477,17 @@ object BenchUtils {
...
+
+          (isGpuPlan(a), isGpuPlan(a.children(i))) match {
+            case (true, true) =>
+              w.println(s"""node$childId -> node$id [color="$nvGreen"];""")
+            case (false, false) =>
+              w.println(s"""node$childId -> node$id [color="$blue"];""")
+            case _ =>
+              // show emphasis on transitions between CPU and GPU
+              w.println(s"node$childId -> node$id [color=red, style=bold];")
+          }
```
Comment (jlowe) [inline integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/BenchUtils.scala:489]:
```suggestion
          val style = (isGpuPlan(a), isGpuPlan(a.children(i))) match {
            case (true, true) => s"color=\"$nvGreen\""
            case (false, false) => s"color=\"$blue\""
            case _ =>
              // show emphasis on transitions between CPU and GPU
              "color=red, style=bold"
           }
           w.println(s"node$childId -> node$id [$style];")
```

**Review by jlowe — DISMISSED** [review]
Minor comments that aren't mustfix, lgtm.

**Review by nartal1 — DISMISSED** [review]

**Comment by andygrove** [comment]
build

**Review by jlowe — DISMISSED** [review]

**Review by nartal1 — DISMISSED** [review]

**Comment by andygrove** [comment]
build

**Comment by andygrove** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**